### ISSUE TITLE
not hide the canera and mute for the remote Participant.

### ISSUE THREAD

**Issue by Mu-Aleem** [twilio/twilio-video.js#2016]
please help me. my hide camera and mute work only for the local Participant not for the remote Participant.

const handleAudioToggle = () => {
room.localParticipant.audioTracks.forEach((track) => {
if (track.track.isEnabled) {
track.track.disable();
} else {
track.track.enable();
}
setToggleAudio(track.track.isEnabled);
});
};

const handleVideoToggle = () => {
room.localParticipant.videoTracks.forEach((track) => {
if (track.track.isEnabled) {
track.track.disable();
} else {
track.track.enable();
}
setToggleVideo(track.track.isEnabled);
});
};

hear is my function.
when i console the room. not find any RemoteAudioTrack or RemoteVideoTrack .

**Comment by Mu-Aleem** [comment]
hi @manjeshbhargav . I'm creating a basic video call app and after connecting the call when I hide or mute my mic its mute and hide on my side only. another user can hire and see my video which is totally wrong. 

this is my code 
const VideoChat = () => {
  const [username, setUsername] = useState("");
  const [roomName, setRoomName] = useState("Twilio_Video_Calling_Room_Name");
  const [room, setRoom] = useState(null);
  const [connecting, setConnecting] = useState(false);
  const [Token, setToken] = useState(null);

  const handleUsernameChange = useCallback((event) => {
    setUsername(event.target.value);
  }, []);

  const handleRoomNameChange = useCallback((event) => {
    setRoomName(event.target.value);
  }, []);

  const handleSubmit = useCallback(
    async (event) => {
      event.preventDefault();
      // setConnecting(true);
      const data = await fetch(
        `https://loquacious-seahorse-693c09.netlify.app/.netlify/functions/api/api/token/${username}`,
        {
          method: "POST",
          body: JSON.stringify({
            identity: username,
            room: roomName,
          }),
          headers: {
            "Content-Type": "application/json",
          },
        }
      ).then((res) => res.json());

      setToken(data.token);
      setRoom(true);
    },
    [roomName, username]
  );

  const handleLogout = useCallback(() => {
    setRoom((prevRoom) => {
      if (prevRoom) {
        prevRoom.localParticipant.tracks.forEach((trackPub) => {
          trackPub.track.stop();
        });
        prevRoom.disconnect();
      }
      return null;
    });
  }, []);

  useEffect(() => {
    if (room) {
      const tidyUp = (event) => {
        if (event.persisted) {
          return;
        }
        if (room) {
          handleLogout();
        }
      };
      window.addEventListener("pagehide", tidyUp);
      window.addEventListener("beforeunload", tidyUp);
      return () => {
        window.removeEventListener("pagehide", tidyUp);
        window.removeEventListener("beforeunload", tidyUp);
      };
    }
  }, [room, handleLogout]);

  let render;
  if (room) {
    render = (
      <Room
        roomName={roomName}
        // room={room}
        token={Token}
        handleLogout={handleLogout}
      />
    );
  } else {
    render = (
      <Lobby
        username={username}
        roomName={roomName}
        handleUsernameChange={handleUsernameChange}
        handleRoomNameChange={handleRoomNameChange}
        handleSubmit={handleSubmit}
        connecting={connecting}
      />
    );
  }
  return render;
};


Room.js 

const Room = ({ roomName, token, handleLogout }) => {
  const [room, setRoom] = useState(null);
  const [participants, setParticipants] = useState([]);

  const [toggleAudio, setToggleAudio] = useState(true);
  const [toggleVideo, setToggleVideo] = useState(true);

  useEffect(() => {
    const participantConnected = (participant) => {
      setParticipants((prevParticipants) => [...prevParticipants, participant]);
    };

    const participantDisconnected = (participant) => {
      setParticipants((prevParticipants) =>
        prevParticipants.filter((p) => p !== participant)
      );
    };

    Video.connect(token, {
      name: roomName,
    }).then((room) => {
      setRoom(room);
      room.on("participantConnected", participantConnected);
      room.on("participantDisconnected", participantDisconnected);
      room.participants.forEach(participantConnected);
    });

    return () => {
      setRoom((currentRoom) => {
        if (currentRoom && currentRoom.localParticipant.state === "connected") {
          currentRoom.localParticipant.tracks.forEach(function (
            trackPublication
          ) {
            trackPublication.track.stop();
          });
          currentRoom.disconnect();
          return null;
        } else {
          return currentRoom;
        }
      });
    };
  }, [roomName, token]);

  const handleCallDisconnect = () => {
    room.disconnect();
  };

  const handleAudioToggle = () => {
    room.localParticipant.audioTracks.forEach((track) => {
      if (track.track.isEnabled) {
        track.track.disable();
      } else {
        track.track.enable();
      }
      setToggleAudio(track.track.isEnabled);
    });
   
 

  const handleVideoToggle = () => {
    room.localParticipant.videoTracks.forEach((track) => {
      if (track.track.isEnabled) {
        track.track.disable();
      } else {
        track.track.enable();
      }
      setToggleVideo(track.track.isEnabled);
    });
  };

  const remoteParticipants = participants.map((participant) => (
    <Participant
      key={participant.sid}
      participant={participant}
      isLocal={false}
    />
  ));

  return (
    <div className="room">
      <h2>Room: {roomName}</h2>
      <button onClick={handleLogout}>Log out</button>
      <div className="local-participant">
        {room ? (
          <Participant
            key={room.localParticipant.sid}
            participant={room.localParticipant}
            handleAudioToggle={handleAudioToggle}
            handleVideoToggle={handleVideoToggle}
            handleCallDisconnect={handleCallDisconnect}
            toggleAudio={toggleAudio}
            toggleVideo={toggleVideo}
            isLocal={true}
          />
        ) : (
          ""
        )}
      </div>
      <h3>Remote Participants</h3>
      <div className="remote-participants">{remoteParticipants}</div>
    </div>
  );
};



Participant.js


const Participant = ({
  key,
  participant,
  handleAudioToggle,
  handleVideoToggle,
  handleCallDisconnect,
  toggleAudio,
  toggleVideo,
  isLocal,
}) => {
 

  const [videoTracks, setVideoTracks] = useState([]);
  const [audioTracks, setAudioTracks] = useState([]);

  const videoRef = useRef();
  const audioRef = useRef();

  const trackpubsToTracks = (trackMap) =>
    Array.from(trackMap.values())
      .map((publication) => publication.track)
      .filter((track) => track !== null);

  useEffect(() => {
    setVideoTracks(trackpubsToTracks(participant.videoTracks));
    setAudioTracks(trackpubsToTracks(participant.audioTracks));

    const trackSubscribed = (track) => {
      if (track.kind === "video") {
        setVideoTracks((videoTracks) => [...videoTracks, track]);
      } else {
        setAudioTracks((audioTracks) => [...audioTracks, track]);
      }
    };

    const trackUnsubscribed = (track) => {
      if (track.kind === "video") {
        setVideoTracks((videoTracks) => videoTracks.filter((v) => v !== track));
      } else {
        setAudioTracks((audioTracks) => audioTracks.filter((a) => a !== track));
      }
    };

    participant.on("trackSubscribed", trackSubscribed);
    participant.on("trackUnsubscribed", trackUnsubscribed);

    return () => {
      setVideoTracks([]);
      setAudioTracks([]);
      participant.removeAllListeners();
    };
  }, [participant]);

  useEffect(() => {
    const videoTrack = videoTracks[0];
    if (videoTrack) {
      videoTrack.attach(videoRef.current);
      return () => {
        videoTrack.detach();
      };
    }
  }, [videoTracks]);

  useEffect(() => {
    const audioTrack = audioTracks[0];
    if (audioTrack) {
      audioTrack.attach(audioRef.current);
      return () => {
        audioTrack.detach();
      };
    }
  }, [audioTracks]);

  return (
    <div className="participant">
      <h3>{participant.identity}</h3>
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      {isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )}
    </div>
  );
};





Please help me to solve this problem

**Comment by Mu-Aleem** [comment]
<div className="participant">
      <h3>{participant.identity}</h3>
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      {isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )}
    </div>

**Comment by Mu-Aleem** [comment]
<div className="participant">
      {/* <h3>{participant.identity}</h3> */}
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      {isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )}
    </div>

**Comment by Mu-Aleem** [comment]
<div className="participant">
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      {isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )}
    </div>

**Comment by Mu-Aleem** [comment]
` <div className="participant">
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      {isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )}
    </div>`

**Comment by Mu-Aleem** [comment]
<div className="participant">
      <video ref={videoRef} autoPlay={true} />
      <audio ref={audioRef} autoPlay={true} />

      isLocal && (
        <div className="">
          <button onClick={handleAudioToggle}>
            {toggleAudio ? "Unmute Audio" : "Mute Audio"}
          </button>
          <button onClick={handleVideoToggle}>
            {toggleVideo ? "View" : "Hide"}
          </button>
        </div>
      )
    </div>

**Comment by Mu-Aleem** [comment]
hi @manjeshbhargav . I'm creating a basic video call app and after connecting the call when I hide or mute my mic its mute and hide on my side only. another user can hire and see my video which is totally wrong.

  const handleAudioToggle = () => {
    room.localParticipant.audioTracks.forEach((track) => {
      if (track.track.isEnabled) {
        track.track.disable();
      } else {
        track.track.enable();
      }
      setToggleAudio(track.track.isEnabled);
    });
    
    
     const handleVideoToggle = () => {
    room.localParticipant.videoTracks.forEach((track) => {
      if (track.track.isEnabled) {
        track.track.disable();
      } else {
        track.track.enable();
      }
      setToggleVideo(track.track.isEnabled);
    });
  };
  
  hear is my function which i use for the hide and mute

**Comment by manjeshbhargav** [comment]
@Mu-Aleem ,

Have you deployed your app somewhere where I can play around with it? Muting is a basic feature that has been working for a long time, so I would like to access your app if possible.
